### PR TITLE
rebuild user module, add more basic information to user

### PR DESCRIPTION
**Pull request by JianFengXxx**
change birthday fields type to Date and add last_login fields
rewrite register/login/getUser endpoint to return more basic information about user

### CODE REVIEW

File: src/main/resources/db/migration/V1.3__update_user_table_add_basic_info_column.sql [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ALTER TABLE user_info
 ADD COLUMN name VARCHAR(20) DEFAULT NULL COMMENT '姓名',
-ADD COLUMN birthday VARCHAR(50) DEFAULT NULL COMMENT '出生年月日',
-ADD COLUMN gender tinyint DEFAULT 1 COMMENT '0-女性，1-男性';
+ADD COLUMN birthday DATE DEFAULT NULL COMMENT '出生年月日',
+ADD COLUMN gender TINYINT DEFAULT 1 COMMENT '0-女性，1-男性',
+ADD COLUMN last_login DATETIME DEFAULT NULL COMMENT '出生年月日';
```
Comment (Zhai-Chao) [inline src/main/resources/db/migration/V1.3__update_user_table_add_basic_info_column.sql]: last_login is corresponding to `出生年月日`?

File: src/main/scala/com/fengjian/blog/common/LocalDateTimeFormatterUtils.scala [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.fengjian.blog.common
+
+import io.circe.Encoder
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+object LocalDateTimeFormatterUtils {
+
+  private val fmt = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
```
Comment (SaiSea) [inline src/main/scala/com/fengjian/blog/common/LocalDateTimeFormatterUtils.scala]:
```suggestion
  private val DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
```

File: src/main/scala/com/fengjian/blog/common/LocalDateTimeFormatterUtils.scala [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  implicit val dateEncoder: Encoder[LocalDateTime] = Encoder.encodeString.contramap[LocalDateTime](_.format(fmt))
```
Comment (SaiSea) [inline src/main/scala/com/fengjian/blog/common/LocalDateTimeFormatterUtils.scala]: If you want to define a util here, it's not a good practice to make it a `implicit` encoder.

File: src/main/scala/com/fengjian/blog/repository/impl/UserRepositoryImpl.scala [hunk 4]
```diff
@@ -3,18 +3,21 @@ package com.fengjian.blog.repository.impl
...
+import java.time.LocalDateTime
 
 
 
 class UserRepositoryImpl(transactor: Resource[IO, HikariTransactor[IO]]) extends UserRepository {
 
   def getUser(id: Int): IO[Either[UserNotFoundError.type , UserPO]] = {
     transactor.use(
-      xa => sql"SELECT id, username, password, nickname FROM user_info where id = $id"
+      xa => sql"SELECT id, username, password, nickname, name, birthday, gender, last_login FROM user_info where id = $id"
```
Comment (SaiSea) [inline src/main/scala/com/fengjian/blog/repository/impl/UserRepositoryImpl.scala:20]:
```suggestion
      xa => sql"SELECT id, user_name, password, nick_name, name, birthday, gender, last_login FROM user_info where id = $id"
```

File: src/main/scala/com/fengjian/blog/router/model/user/UserCreateDTO.scala [hunk 5]
```diff
@@ -2,8 +2,13 @@ package com.fengjian.blog.router.model.user
...
-case class UserCreateDTO(username: String, password: String, nickname: String) {
 
-  def convert2PO(): UserPO = UserPO(None, username, password, nickname)
+case class UserCreateDTO(username: String, password: String, nickname: String, name: String, birthday: String, gender: Int) {
+
+  def convert2PO(): UserPO = {
+    UserPO(None, username, password, nickname, name, LocalDate.parse(birthday), gender, LocalDateTime.now())
+  }
+
 }
```
Comment (SaiSea) [inline src/main/scala/com/fengjian/blog/router/model/user/UserCreateDTO.scala]: empty line

File: src/main/scala/com/fengjian/blog/service/model/UserInfoResponse.scala [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        "nickName" -> Json.fromString(userInfo.nickname),
+        "name" -> Json.fromString(userInfo.name),
+        "birthday" -> userInfo.birthday.asJson,
+        "gender" -> Json.fromInt(userInfo.gender),
+        "lastLogin" -> userInfo.lastLogin.asJson
+      )
+    }
+  }
+
+}
```
Comment (SaiSea) [inline src/main/scala/com/fengjian/blog/service/model/UserInfoResponse.scala]: empty line

**Pull request merged**